### ISSUE TITLE
[Bug]: Search failed reporting "User Field(int64) is not loaded"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#22902]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus master latest
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    rocksmq
- SDK version(e.g. pymilvus v2.0.0rc2):2.3.0.dev48
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed reporting "User Field(int64) is not loaded"

https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-22900/1/pipeline/110

```

[2023-03-21T09:55:56.073Z] [2023-03-21 09:55:43 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.030655232972244484, 0.7355428921361702, 0.18271045376929385, 0.060465871864248566, 0.5510853038749232, 0.9373964439723786, 0.4597191950920263, 0.18628576535079955, 0.9930028778771359, 0.9188181866667626, 0.7409411613334659, 0.3459241870111952, 0.729170966628837, 0.19578078196268012, 0.210735154......, kwargs: {'_async': False} (api_request.py:56)

[2023-03-21T09:55:56.073Z] [2023-03-21 09:55:44 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=1, reason=Search 1 failed, reason [UnexpectedError] Assert "User Field(int64) is not loaded" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:491

[2023-03-21T09:55:56.073Z]  err %!w(<nil>))>, <Time:{'RPC start': '2023-03-21 09:55:43.868669', 'RPC error': '2023-03-21 09:55:44.358731'}> (decorators.py:108)

[2023-03-21T09:55:56.073Z] [2023-03-21 09:55:44 - ERROR - ci_test]: Traceback (most recent call last):

[2023-03-21T09:55:56.073Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper

[2023-03-21T09:55:56.073Z]     res = func(*args, **_kwargs)

[2023-03-21T09:55:56.073Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request

[2023-03-21T09:55:56.073Z]     return func(*arg, **kwargs)

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 660, in search

[2023-03-21T09:55:56.073Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler

[2023-03-21T09:55:56.073Z]     raise e

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler

[2023-03-21T09:55:56.073Z]     return func(*args, **kwargs)

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler

[2023-03-21T09:55:56.073Z]     ret = func(self, *args, **kwargs)

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler

[2023-03-21T09:55:56.073Z]     raise e

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler

[2023-03-21T09:55:56.073Z]     return func(self, *args, **kwargs)

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 518, in search

[2023-03-21T09:55:56.073Z]     return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests

[2023-03-21T09:55:56.073Z]     raise pre_err

[2023-03-21T09:55:56.073Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests

[2023-03-21T09:55:56.073Z]     raise MilvusException(response.status.error_code, response.status.reason)

[2023-03-21T09:55:56.073Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=1, reason=Search 1 failed, reason [UnexpectedError] Assert "User Field(int64) is not loaded" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:491

[2023-03-21T09:55:56.073Z]  err %!w(<nil>))>

[2023-03-21T09:55:56.073Z]  (api_request.py:39)

[2023-03-21T09:55:56.073Z] [2023-03-21 09:55:44 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=1, reason=Search 1 failed, reason [UnexpectedError] Assert "User Field(int64) is not loaded" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:491
```

### Expected Behavior

Search successfully

### Steps To Reproduce

In CI:
https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20HA%20CI/detail/PR-22900/1/pipeline/110

```python
    @pytest.mark.tags(CaseLabel.L1)
    @pytest.mark.parametrize("expression", cf.gen_normal_expressions())
    def test_search_with_expression(self, dim, expression, _async):
        """
        target: test search with different expressions
        method: test search with different expressions
        expected: searched successfully with correct limit(topK)
        """
        # 1. initialize with data
        nb = 1000
        collection_w, _vectors, _, insert_ids = self.init_collection_general(prefix, True,
                                                                             nb, dim=dim,
                                                                             is_index=False)[0:4]

        # filter result with expression in collection
        _vectors = _vectors[0]
        expression = expression.replace("&&", "and").replace("||", "or")
        filter_ids = []
        for i, _id in enumerate(insert_ids):
            int64 = _vectors.int64[i]
            float = _vectors.float[i]
            if not expression or eval(expression):
                filter_ids.append(_id)

        # 2. create index
        index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
        collection_w.create_index("float_vector", index_param)
        collection_w.load()

        # 3. search with expression
        log.info("test_search_with_expression: searching with expression: %s" % expression)
        vectors = [[random.random() for _ in range(dim)] for _ in range(default_nq)]
        search_res, _ = collection_w.search(vectors[:default_nq], default_search_field,
                                            default_search_params, nb, expression,
                                            _async=_async,
                                            check_task=CheckTasks.check_search_results,
                                            check_items={"nq": default_nq,
                                                         "ids": insert_ids,
                                                         "limit": min(nb, len(filter_ids)),
                                                         "_async": _async})
```

### Milvus Log

[artifacts-milvus-standalone-PR-22900-1-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20HA%20CI/job/PR-22900/1/artifact/artifacts-milvus-standalone-PR-22900-1-pymilvus-e2e-logs.tar.gz)

**Comment by binbinlv** [comment]
Reproduced in nightly, so set it as critical.

**Comment by smellthemoon** [comment]
It seems like can be fixed with #22938.

**Comment by smellthemoon** [comment]
@binbinlv plz check it
